oper8/controller.py

```diff
@@ -168,6 +168,22 @@ def after_deploy(self, session: Session) -> bool:
         """
         return True
 
+    def after_deploy_unsuccessful(self, session: Session) -> bool:
```

Comment:
I haven't made it to the implementation yet, but I think it might be worth adding an additional argument to indicate whether the lack of success was a failure or an early termination. Something like `failed: bool` that would be populated with the `phase1_failed` value.

Reply:
```suggestion
    def after_deploy_unsuccessful(self, session: Session, failed: bool) -> bool:
```

Reply:
I see. 

So, currently only `phase1_completed` is used to judge whether to trigger `after_deploy_unsuccessful` at [here](https://github.com/IBM/oper8/pull/131/files#diff-fab88ac3bcb33457972b94f4b89a3fdb64b1ba7bfdb3a686db262cd1c705f447R325). But when `phase1_completed` is `False`, `phase1_failed` could be either `True` (deployment is failing) or `False` (deployment is incomplete), and without your suggested `failed` argument, user cannot know which is the case.

Thank you for the suggested commits. I will apply them.